core/Period/Factory.php

```diff
@@ -55,6 +56,17 @@ public static function build($period, $date, $timezone = 'UTC')
                 return new Year($date);
                 break;
         }
+
+        $customPeriodFactories = StaticContainer::get('period.custom.factories');
```

Comment:
I don't think it would make sense to ever remove a day period. Would move it to eg `CoreHome` plugin so it cannot be disabled. The advantage is that we would notice if something breaks in the code for "custom periods" as the "core" is using it itself and be good to have them as examples in a plugin.